responder/api.py

```diff
@@ -300,6 +305,30 @@ def no_response(req, resp, **params):
 
         return resp
 
+    def add_event_handler(self, event_type, handler, seconds=None):
+        """Add a event handler to the API.
+
+        :param event_type: A string in ("startup", "cleanup", "shutdown", "tick")
+        :param handler: The function to run. Can be either a function or a coroutine
+        :param seconds: Used when event_type=="tick", as time between each tick
+        """
+
+        if event_type.lower() == "tick":
+            assert seconds != None
+            async def ticker(handler, seconds):
```

Comment:
Couple of things here: really you need to co-ordinate timer events to stop properly on shutdown. (And only complete shutdown once any pending timer event completes.) Also the sync calls ought to happen within a threadpool (there’s other places where this needs to happen too)

I’d consider perhaps raising a PR just for startup/shutdown first, then looking at timer events separately.

(I’ll be implementing the same for Starlette next week, which might be helpful here.)